lib/countly.d.ts

```diff
@@ -0,0 +1,516 @@
+declare module "countly-sdk-web" {
+  namespace Countly {
+    /**
+     * It initializes the SDK and exposes it to the window object. It should be called before any other SDK methods in sync implementations.
+     * @param {object} config - Configuration object. Determines the SDK behavior. Some essential keys are:
+     *
+     * app_key : "app key" from your Countly server application. (MANDATORY!)
+     *
+     * url : Your Countly server URL. You may also use your own server URL or IP here. (MANDATORY!)
+     *
+     * debug : Enables SDK logs to be printed into the browser console. (default: false)
+     *
+     * For the full list of configuration options, see https://support.count.ly/hc/en-us/articles/360037441932-Web-analytics-JavaScript#h_01HABTQ439HZN7Y6A6F07Y6G0K
+     */
+    function init(config: object): void;
+
+    /**
+     * Modify feature groups for consent management. Allows you to group multiple features under one feature group
+     * @param {object} features - object to define feature name as key and core features as value
+     * @example <caption>Adding all features under one group</caption>
+     * Countly.group_features({all:["sessions","events","views","scrolls","clicks","forms","crashes","attribution","users"]});
+     * //After this call Countly.add_consent("all") to allow all features
+     * @example <caption>Grouping features</caption>
+     * Countly.group_features({
+     *    activity:["sessions","events","views"],
+     *    interaction:["scrolls","clicks","forms"]
+     * });
+     * //After this call Countly.add_consent("activity") to allow "sessions","events","views"
+     * //or call Countly.add_consent("interaction") to allow "scrolls","clicks","forms"
+     * //or call Countly.add_consent("crashes") to allow some separate feature
+     */
+    function group_features(features: object): void;
+
+    /**
+     * Check if consent is given for specific feature (either core feature of from custom feature group)
+     * @param {string} feature - name of the feature, possible values, "sessions","events","views","scrolls","clicks","forms","crashes","attribution","users" or custom provided through {@link Countly.group_features}
+     * @returns {Boolean} true if consent was given for the feature or false if it was not
+     */
+    function check_consent(feature: string): boolean;
+
+    enum DeviceIdType {
+      DEVELOPER_SUPPLIED,
+      SDK_GENERATED,
+      TEMPORARY_ID,
+    }
+
+    /**
+     * Countly SDK async queue
+     * You can push methods here to be executed after Countly SDK is fully loaded
+     * @example
+     * Countly.q.push(['track_sessions']);
+     * Countly.q.push(['add_event',{
+     *   "key": "click",
+     *   "count": 1,
+     *   "sum": 1.5,
+     *   "dur": 30,
+     *   "segmentation": {
+     *     "key1": "value1",
+     *     "key2": "value2"
+     *   }
+     * }]);
+     * @type {Array}
+     */
+    let q: any[];
```

Comment:
As a side note, we are now trying to minimize parameter names.
But this leads to reduced code readability.

Reply:
This thought is not related to this "q" but, related to all SDKs

Reply:
ofc. this one is unfortunately unreachable at this point